service/ExchangeService.js

```diff
@@ -193,88 +327,208 @@ var fetchFromExmo= function(callback){
+			callback(null, data);
+			return;
+		});
 	});
+
 };
 
-var fetchFromBithumb= function(callback){
-	fetchExchangeData(AppConfig.exchanges.yahoofinance.name, AppConfig.exchanges.yahoofinance.url, function(err, result){
+var fetchFromCexio= function(callback){
+	var data = {};
+	log.debug('Fetching exchange data from ' + AppConfig.exchanges.cexio.name + ' at ' + AppConfig.exchanges.cexio.url);
+	fetchExchangeData(AppConfig.exchanges.cexio.name, AppConfig.exchanges.cexio.url, function(err, result){
 		if ( err ){
 			callback(err);
-		}else{
-			krwUsd = result.query.results.rate.Rate;
-			fetchExchangeData(AppConfig.exchanges.bithumb.name, AppConfig.exchanges.bithumb.url, function(err, result){
+			return;
+		}
+		if (result === null) {
+			callback(null, null);
+			return;
+		}
+		data = {
+			exchange: AppConfig.exchanges.cexio.name,
+			url: AppConfig.exchanges.cexio.orgUrl,
+			price: parseFloat(result.lprice)
+		};
+		fetchExchangeData(AppConfig.exchanges.cexio.name, AppConfig.exchanges.cexio.urlEur, function(err, result){
+			if ( err ){
+				callback(err);
+				return;
+			}
+			if (result === null) {
+				callback(null, null);
+				return;
+			}
+			data['price_eur'] = parseFloat(result.lprice);
+			fetchExchangeData(AppConfig.exchanges.cexio.name, AppConfig.exchanges.cexio.urlBTC, function(err, result){
 				if ( err ){
 					callback(err);
-				}else{
-					var data = {
-						exchange: AppConfig.exchanges.bithumb.name,
-						url: AppConfig.exchanges.bithumb.orgUrl,
-						price: result.data.closing_price*krwUsd,
-					};
-					data.volume = result.data.volume_1day*data.price;
-					callback(null, data);
+					return;
 				}
+				if (result === null) {
+					callback(null, null);
+					return;
+				}
+				data['price_btc'] = parseFloat(result.lprice);
+				callback(null, data);
+				return;
 			});
+		});
+
+	});
+
+
+};
+
+var fetchFromBithumb= function(callback){
+	fetchExchangeData(AppConfig.exchanges.yahoofinance.name, AppConfig.exchanges.yahoofinance.url, function(err, result){
+		if ( err ){
+			callback(err);
+			return;
+		}
+		if (result === null || result.query.results === null) {
+			callback(null, null);
+			return;
 		}
+		krwUsd = parseFloat(result.query.results.rate.Rate);
+		fetchExchangeData(AppConfig.exchanges.bithumb.name, AppConfig.exchanges.bithumb.url, function(err, result){
+			if ( err ){
+				callback(err);
+				return;
+			}
+			if (result === null) {
+				callback(null, null);
+				return;
+			}
+			var data = {
+				exchange: AppConfig.exchanges.bithumb.name,
+				url: AppConfig.exchanges.bithumb.orgUrl,
+				price: result.data.closing_price*krwUsd
+			};
+			data.volume = result.data.volume_1day*data.price;
+			callback(null, data);
+			return;
+		});
 	});
 
 
 };
 
+var addUsdPrice = function addUsdPrice(arr){
+	updatedResult = arr.filter(function(obj) {
+		if (!obj){
+			return false;
+		}
+		if(obj && obj.hasOwnProperty("price_btc") && !obj.hasOwnProperty("price")){
```

Comment:
we filtered above all empty items (`!obj`), so we don't need to check it anymore:
```
if(obj.hasOwnProperty("price_btc") && !obj.hasOwnProperty("price")){
  ...
}
return true;
```